### DISCUSSION TITLE
Getting non-nodal values to output with nodal values in same .csv or vice versa

### DISCUSSION THREAD

**Discussion by jackeggemeyer** [idaholab/moose#27524] [Q&A General] [answered]
I currently use VectorPostprocessors to output some of my nodal variables to a .csv file. I want my output to have a variable per each node and timestep. However, I also want to output an elemental variable for each node.

My elemental variable must stay elemental as it is part of the discrete nucleation kernel and has to be an elemental auxiliary variable. Is there a way to get nucl_map to output with the vectorpostprocessors? Like an average of the elemental values per node? I looked at the DiscreteNucleatoinData post-processor. I provided some code below. Thank you so much for your help! Let me know if you need more information.
```
[AuxVariables]
  [./nucl_map]
     order = first
     family = MONOMIAL
     outputs = exodus
  [../]
[]

[AuxKernels]
  [./nucl_map_aux]
    type = DiscreteNucleationAux
    map = map
    variable = nucl_map
    no_nucleus_value = 0
    nucleus_value = 1
  [../]
[]

[VectorPostprocessors]
    [./etarad]
    type = NodalValueSampler
    variable = 'etab0 etam0'
    sort_by = 'id'
  [../] 
[] 
```
I also wouldn't mind the nodal variables being ouputed per element either I just wanted the data synced.

**Comment by GiudGiud** [comment, marked as answer]
You could project the elemental variable on a nodal one with the ProjectionAux, then output that one
https://mooseframework.inl.gov/source/auxkernels/ProjectionAux.html
